frontend/exporter/src/traits.rs

```diff
@@ -376,14 +376,30 @@ impl<'tcx> IntoImplExpr<'tcx> for rustc_middle::ty::PolyTraitRef<'tcx> {
     }
 }
 
-pub fn clause_id_of_predicate(predicate: rustc_middle::ty::Predicate) -> u64 {
+#[tracing::instrument(level = "trace", skip(s))]
+pub fn clause_id_of_predicate<'tcx, S: UnderOwnerState<'tcx>>(
```

Comment:
So you are attaching a hash here to make the ids unique? Please add documentation what is happening and why.